ERCS/erc-7589.md

```diff
@@ -0,0 +1,480 @@
+  [ERC-1155](./eip-1155.md) approvals. More information can be found in the [Role Approvals](#role-approvals) section.
+
+## Rationale
+
+The conventional methods for utilizing SFTs often involve transferring tokens to another address or freezing balances to
+restrict transfers. However, the question arises: How can SFT owners securely delegate the utility of their assets
+without compromising custody?
+
+[ERC-7589](./eip-7589.md) steps in as a solution by introducing the concept of "token commitments". This abstraction
+serves as a powerful tool for users looking to delegate the control of their SFTs. A token commitment can represent
+either a frozen balance or tokens deposited into a contract, offering a standardized and secure way for SFT owners to
+delegate the use of their assets. Through ERC-7589, users gain a versatile mechanism to abstract the complexities of
+secure delegation, enhancing the utility and interoperability of semi-fungible tokens.
+
+[ERC-7589](./eip-7589.md) IS NOT an extension of [ERC-1155](./eip-1155.md). The main reason behind this decision is to
+keep the standard agnostic of any implementation. This approach enables the standard to be implemented externally or on
+the same contract as the SFT and allows dApps to use roles with immutable SFTs.
+
+### Role Approvals
+
+Like [ERC-1155](./eip-1155.md), [ERC-7589](./eip-7589.md) allows users to approve operators to grant and revoke roles on
+their behalf. This feature is crucial for interoperability, as it enables third-party applications to manage user roles
+without custody-level approvals. Role approvals are part of the core interface, and compliant contracts MUST implement
+the `setRoleApprovalForAll` and `isRoleApprovedForAll` functions.
+
+### Automatic Expiration
+
+Automatic expiration is implemented to save users gas. To end a role assignment, instead of requiring users always to
+call `revokeRole`, applications SHOULD call the `roleExpirationDate` and compare it to the current timestamp to check if
+the role is still valid.
+
+In the context of [ERC-7589](./eip-7589.md), dates are represented as `uint64`. The maximum UNIX timestamp represented
+by a `uint64` is about the year 584 billion, which should be enough to be considered "permanent". For this reason, it's
+RECOMMENDED using `type(uint64).max` in use cases where the assignment is required never to expire.
+
+### Revocable Roles
+
+In certain scenarios, the grantor might need to revoke a role before its expiration. While in others, the grantee
+requires assurance that the role can't be prematurely revoked (e.g. when the grantee pays tokens to utilize them). The
+`_revocable` parameter was included in the `grantRole` function for this exact reason, and it specifies whether the
+grantor can revoke the role prior to the expiration date. Regardless of the `_revocable` value, the grantee SHOULD
+always be able to revoke roles, allowing recipients to eliminate undesirable assignments.
```

Comment:
Similarly, SHOULD here is introducing a requirement, so that sentence should be in the specification section.